backends/k8s/client.go

```diff
@@ -0,0 +1,278 @@
+	"k8s.io/client-go/pkg/runtime/schema"
+	"k8s.io/client-go/pkg/runtime/serializer"
+	"k8s.io/client-go/pkg/runtime"
+	"github.com/kelseyhightower/confd/log"
+	"github.com/projectcalico/libcalico-go/lib/backend/k8s/thirdparty"
+	kapiv1 "k8s.io/client-go/pkg/api/v1"
+	metav1 "k8s.io/client-go/pkg/apis/meta/v1"
+	clientapi "k8s.io/client-go/pkg/api"
+	kerrors "k8s.io/client-go/pkg/api/errors"
+	"regexp"
+	"github.com/projectcalico/libcalico-go/lib/backend/k8s/resources"
+	"github.com/projectcalico/libcalico-go/lib/backend/model"
+	"strconv"
+)
+
+const (
+	ipPool          = "/calico/v1/ipam/v4/pool"
+	global          = "/calico/bgp/v1/global"
+	globalASN       = "/calico/bgp/v1/global/as_num"
+	globalNodeMesh  = "/calico/bgp/v1/global/node_mesh"
+	allNodes        = "/calico/bgp/v1/host"
+	globalLogging   = "/calico/bgp/v1/global/loglevel"
+)
+
+var (
+	singleNode = regexp.MustCompile("^/calico/bgp/v1/host/([a-zA-Z0-9._/-]*)")
+)
+
+type Client struct {
+	clientSet *kubernetes.Clientset
+	tprClient *rest.RESTClient
+}
+
+func NewK8sClient(kubeconfig string) (*Client, error) {
+
+	log.Debug("Building k8s client")
+
+	// Set an explicit path to the kubeconfig if one
+	// was provided.
+	loadingRules := clientcmd.ClientConfigLoadingRules{}
+	if kubeconfig != "" {
+		log.Debug(fmt.Sprintf("Using kubeconfig: \n%s", kubeconfig))
+		loadingRules.ExplicitPath = kubeconfig
+	}
+
+	// A kubeconfig file was provided.  Use it to load a config, passing through
+	// any overrides.
+	config, err := clientcmd.NewNonInteractiveDeferredLoadingClientConfig(
+		&loadingRules, &clientcmd.ConfigOverrides{}).ClientConfig()
+	if err != nil {
+		return nil, err
+	}
+
+	// Create the clientset
+	cs, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		return nil, err
+	}
+	log.Debug(fmt.Sprintf("Created k8s clientSet: %+v", cs))
+
+	tprClient, err := buildTPRClient(config)
+	if err != nil {
+		return nil, err
+	}
+	kubeClient := &Client{
+		clientSet: cs,
+		tprClient: tprClient,
+	}
+
+    	return kubeClient, nil
+}
+
+// Take the etcd like keys and route it to the appropriate k8s API endpoint
+func (c *Client) GetValues(keys []string) (map[string]string, error) {
+	var kvps = make(map[string]string)
+	for _, key := range keys {
+		log.Debug(fmt.Sprintf("Getting key %s", key))
+		if m := singleNode.FindStringSubmatch(key); m != nil {
+			host := m[len(m)-1]
+			kNode, err := c.clientSet.Nodes().Get(host, metav1.GetOptions{})
+			if err != nil {
+				return nil, err
+			}
+			err = populateNodeDetails(kNode, &kvps)
+			if err != nil {
+				return nil, err
+			}
+		}
+		switch key {
+		case global:
+			// Default to "info" until this makes it into k8s
+			kvps[globalLogging] = "info"
+			// Default to 64512
+			kvps[globalASN] = "64512"
+			// Default to False until peering info is available in k8s
+			kvps[globalNodeMesh] = `{"enabled": true}`
+		case globalNodeMesh:
+			// This is needed as there are calls to just 'global' and 'global/loglevel'
```

Comment:
Ah sorry no I misread what you said, the comment is incorrect, that should read global/nodemesh not loglevel as you are correct I don't believe there are any direct calls to loglevel